### ISSUE TITLE
QC-CV -> MS-CV

### ISSUE THREAD

**Issue by mwalzer** [HUPO-PSI/psi-ms-CV#80]
### Describe the question or discussion

Dear PSI-MS CV community,  
_( I posted this message to the PSI-MS-CV mailing list, but is stuck in the approval queue. Sorry for the duplication, GitHub issues is a better place to have such discussions in any case )_

I am writing on behalf of the PSI-QC working group, as we want to get our QC-CV 'into the fold'. Before that, however, we thought it'd be best to start a conversation on how. 
We have been developing a separate CV initially to have some freedoms on what/how things work out best for the mzQC format, which relies heavily on CV control. 
Our CV holds the metric definitions, relations and few additional 'organisational' terms, so far <300 entries, for which we chose the QC prefix (naturally). We ditched the `xref: value-type` declarations in favour of defining term values with full CV terms and relations, which works nicely with the format. 
The mzQC format is, unlike most PSI formats, JSON based. One of the goals in the design was to bring the format closer to the last miles of analysis (R/pandas/web) but also as something that works between any steps of an MS-based study workflow (lightweight with high automation potential). If you want to know more, check out our specification document draft ( https://hupo-psi.github.io/mzQC/). For mzQC, we have to cover a vast range of QC/MS-related topics in the CV. For that, we've devised a set of templates and rules for adding new term/metrics via GitHub issues. To keep better track of the entries, Nils created a GitHub workflow for obo-to-markup conversion to enhance browsability. QC-CV is not standalone, we base as much as possible on existing definitions, so we imported few other CVs: UO, OBI, STATO (,NCIT) and PSI-MS of course. So, overall I think QC-CV would fit in nicely, either as big PR or piece-wise. As for the merge, maybe we can follow the same playbook as the PEFF integration did? What do you think? 

**TL;DR**: mzQC folks want to merge their CV into PSI-MS. But before opening a gargantuan PR, we should talk. 

Best,

**Comment by edeutsch** [comment]
Hi Mathias, thanks for contacting us. Your message did actually go out to the list quite rapidly, but you didn't get a notification that it went out, just that it was briefly held. so everyone has seen it. We're just slow.

We should definitely set up a conversation about this. A few comments from me:
- Where can we have a look at what you have so far?
- Shall we set up a doodle poll to select a time when interested parties can meet?
- What is your level of urgency? It is high holiday time now
- I think I would dissuade retaining the QC prefix unless your term names don't follow MS policy (term name like you would use it in a sentence; PEFF keywords do not)
- Not using the MS value-type system is also a bit concerning, but we should examine what you did

Those are my thoughts. Let's find a time to talk!

**Comment by FarmGeek4Life** [comment]
It looks like this is probably their current version: https://github.com/HUPO-PSI/mzQC/blob/main/cv/qc-cv.obo

**Comment by bittremieux** [comment]
Indeed, that's our current QC CV. We still have a few term requests in the issues that we will include the finalize the CV before folding it into the MS CV.

I think setting up a poll to find a time to discuss it would be a good next step. Merging the CV is the final thing to take care of before mzQC can proceed to the next review step, but we've been in development for such a long time that a two week delay because of the holidays is not going to be the end of the world either. We should preferably not overly delay the call when most people are available again though.
Most of our working group members are in Europe, so a morning time slot PDT is probably going to work best.

**Comment by chambm** [comment]
I understand the value-type system wasn't suitable for n-tuple, table, and matrix type terms, which I suppose are represented in JSON strings. Can we simply use string as the value-type for those kind of terms? We could use a type-specific regex instead (e.g. validate n-table vs table), but that would probably be overkill. The current is_a relationship to those term types should be fine to keep though.

**Comment by mobiusklein** [comment]
We recently added a `has_value_type` relationship which replicates the existing `xref: value-type:...` mechanism, but use another entity as the object of the predicate instead of just the built-in XSD types, similar to what you're doing. Then the representation of the value in the "document" is dependent upon the document format (XML, JSON, plain text, binary). At least, that's the approach we converged on for mzSpecLib.

The drawback is that we are then leaving how you render a term-defined type uncontrolled. It looks like mzQC approached it by specifying the term in the CV but defining its meaning in a specification document:

https://github.com/HUPO-PSI/mzQC/blob/7a362c7eef15023a69ebcb3f9458e3fbe723521a/cv/qc-cv.obo#L74-L78

I am of mixed opinion on this because I like the idea, but it mixes description of an entity with data structure/layout. That moves meaning out of the document schema and into the CV, when the document is what gives the CV context for interpretation. Then again, allowing more composability isn't a bad thing, it's just important that it be well defined. I recently learned I've been misusing a CV term because I saw an example of it where it was used incorrectly early on, and without a "good" counter-example I propagated that pattern throughout my work.

For validation, we could define a JSONSchema for each metric value type and put it at a stable URL for the specification to reference?

**Comment by mwalzer** [comment]
I just started a doodle to find a date. See mailinglist/slack for link and details.

**Comment by edeutsch** [comment]
There are still some more terms to come, but it seems like the process is well underway and this issue can be closed. Reopen if necessary.
